Feature01

hasi-b opened the pull request:
For checking

hasi-b committed:
POL-18

Player State Added and bulet state added

hasi-b committed:
POL-18

Player health change system based on switching

hasi-b committed:
POL-18

Bullet Script changes.
Fixed the issue of  bullet destroying right after spawning

hasi-b committed:
POL-18

naming convention fixed

hasi-b committed:
POL-18

Health Manager added

Warhammer4000 commented on Assets/Feature01/Scripts/Bullet/BlueBullet.cs:
```diff
@@ -0,0 +1,15 @@
+
+public class BlueBullet : BulletAbstractClasses
+{
+    Color BlueColorBullet = Color.blue;
```
Wrong. Color property should be on Abstract class. It should just be called Color. Not Blue Color or red color. If you don't use generic name you won't be able to use generic codes.

Warhammer4000 commented on Assets/Feature01/Scripts/Bullet/BlueBullet.cs:
```diff
@@ -0,0 +1,15 @@
+
+    public override void InitializeBullet(GameObject bullet)
+    {
+        bullet.GetComponent<MeshRenderer>().sharedMaterial.color = BlueColorBullet;
```
Same issue here. It should be initialized from the abstract base class with the color. InitializeBullet should be a virtual method which can be overridden if needed.

Warhammer4000 commented on Assets/Feature01/Scripts/Bullet/GreenBullet.cs:
```diff
@@ -4,12 +4,12 @@
 
 public class GreenBullet : BulletAbstractClasses
```
Same issue as previous

Warhammer4000 commented on Assets/Feature01/Scripts/Bullet/RedBullet.cs:
```diff
@@ -4,11 +4,11 @@
 
 public class RedBullet : BulletAbstractClasses
```
Same issue

Warhammer4000 commented on Assets/Feature01/Scripts/CharacterStates/PolarityManager.cs:
```diff
@@ -0,0 +1,84 @@
+
+    private void Awake()
+    {
+        PlayerStatesSelection =  new Dictionary<PlayerState, Stats>()
```
This is okay. But you should probably drag drop them from the inspector instead of using resources folder like this, Then again this could be fine in the long run.

Warhammer4000 commented on Assets/Feature01/Scripts/CharacterStates/PolarityManager.cs:
```diff
@@ -0,0 +1,84 @@
+
+        MaxHealth = CurrentPlayerStats.MaxHealth;
+        MovementSpeed = CurrentPlayerStats.MovementSpeed;
+        CurrentHealth = 100f;
```
You should probably have a method to set the health. As you know the max health is a limiting condition. What happens if max hp is 75 and you are setting 100f ?

Warhammer4000 replied:
You should probably do this from the health manager class

Warhammer4000 commented on Assets/Feature01/Scripts/Controller/PlayerController.cs:
```diff
@@ -52,9 +53,9 @@ private void FixedUpdate()
+        movementController.PlayerMove(movementInput,PolarityManager.MovementSpeed,_rigidbody);
 
-        movementController.PlayerLook(getDirection, _rigidbody, this.transform);
+        movementController.PlayerLook(GetDirection, _rigidbody, this.transform);
```
not necessary to add this

Warhammer4000 commented on Assets/Feature01/Scripts/Health/HealthManager.cs:
```diff
@@ -0,0 +1,40 @@
+public class HealthManager : MonoBehaviour
+{
+
+
```
I think you should send in the stats inside here. Inject it. Because it will change as your states change, then use the add remove health from this class. So finally, the stats will always represent the data you need to see

Warhammer4000 commented on Assets/Feature01/Scripts/Health/HealthManager.cs:
```diff
@@ -0,0 +1,40 @@
+
+
+
+    public float CalculateCurrentHealth(PolarityManager.PlayerState oldState,PolarityManager.PlayerState newState, float _currentHealth,Dictionary<PolarityManager.PlayerState,Stats> PlayerStatesSelection)
```
This got complicated as you didn't bring in the stats here. 
Here's my suggestion. Have two private stats variable here. 
1) Current Stats
2) Previous Stats

Switch them out when stats change. Once you do it. these methods can be simplified.

Warhammer4000 reviewed (changes requested):
Please make the changes

hasi-b committed:
POL-18

Changed the HealthManager and added two states

hasi-b committed:
POL-18

Previous issue with bullets fixed,
Used List instead of Dictionary in the PolarityManager

hasi-b committed:
POL-17

Bullet System

hasi-b committed:
POL-17

FieldOfview script
